src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -189,9 +255,34 @@ public static TypedData ToRpc(this object value, ILogger logger, Capabilities ca
                     }
                 }
             }
-            else
+            return typedData;
+        }
+
+        public static TypedData ToRpc(this object value, ILogger logger, Capabilities capabilities)
+        {
+            TypedData typedData = null;
+            if (value == null)
+            {
+                typedData = new TypedData();
+                return typedData;
+            }
+            typedData = value.ToRpcPrimitive();
+            if (typedData == null)
+            {
+                typedData = value.ToRpcPrimitive();
+            }
+            if (typedData == null)
+            {
+                typedData = value.ToRpcHttp(logger, capabilities);
+            }
+            if (typedData == null && IsTypeDataCollectionSupported(capabilities))
```

Comment:
Check for IsTypeDataCollectionSupported first. Move ToPrimitive and ToRpc to else block.

Reply:
Added to code to address the enhancements.